stdlib/3/io.pyi

```diff
@@ -77,52 +76,27 @@ class FileIO(RawIOBase):
         opener: Optional[Callable[[Union[int, str], str], int]] = ...
     ) -> None: ...
 
-# TODO should extend from BufferedIOBase
-class BytesIO(BinaryIO):
+class BytesIO(BufferedIOBase, BinaryIO):
     def __init__(self, initial_bytes: bytes = ...) -> None: ...
     # BytesIO does not contain a "name" field. This workaround is necessary
     # to allow BytesIO sub-classes to add this field, as it is defined
     # as a read-only property on IO[].
     name: Any
+    def __enter__(self: _T) -> _T: ...
     def getvalue(self) -> bytes: ...
     def getbuffer(self) -> memoryview: ...
-    # copied from IOBase
-    def __iter__(self) -> Iterator[bytes]: ...
-    def __next__(self) -> bytes: ...
-    def __enter__(self) -> BytesIO: ...
-    def __exit__(self, t: Optional[Type[BaseException]] = ..., value: Optional[BaseException] = ...,
-                 traceback: Optional[TracebackType] = ...) -> Optional[bool]: ...
-    def close(self) -> None: ...
-    def fileno(self) -> int: ...
-    def flush(self) -> None: ...
-    def isatty(self) -> bool: ...
-    def readable(self) -> bool: ...
-    def readlines(self, __size: int = ...) -> List[bytes]: ...
-    def seek(self, __pos: int, __whence: int = ...) -> int: ...
-    def seekable(self) -> bool: ...
-    def tell(self) -> int: ...
-    def truncate(self, __size: Optional[int] = ...) -> int: ...
-    def writable(self) -> bool: ...
-    # TODO should be the next line instead
-    # def writelines(self, lines: List[Union[bytes, bytearray]]) -> None: ...
-    def writelines(self, __lines: Any) -> None: ...
-    def readline(self, __size: Optional[int] = ...) -> bytes: ...
-    def __del__(self) -> None: ...
-    closed: bool
-    # copied from BufferedIOBase
-    def detach(self) -> RawIOBase: ...
-    def readinto(self, __buffer: _bytearray_like) -> int: ...
-    def write(self, __b: Union[bytes, bytearray]) -> int: ...
-    def readinto1(self, __buffer: _bytearray_like) -> int: ...
-    def read(self, __size: Optional[int] = ...) -> bytes: ...
     if sys.version_info >= (3, 7):
-        def read1(self, __size: Optional[int] = ...) -> bytes: ...
+        def read1(self, __size: int = ...) -> bytes: ...
```

Comment:
Actually this does accept `None`, so it should have stayed Optional. Just applied this change myself. If Travis is happy now I'll merge this.